src/chainrules.jl

```diff
@@ -0,0 +1,19 @@
+function rrule(::typeof(*), A::LinearMap, x::AbstractVecOrMat)
+    y = A*x
+    function pullback(dy)
+      DY = unthunk(dy)
+      # Because A is an abstract map, the product is only differentiable w.r.t the input
+      return NoTangent(), NoTangent(), @thunk(A' * DY)
```

Comment:
This will work only for maps that have an adjoint.
To help out users, I recommend a try/catch like we have for `getindex` here
https://github.com/JuliaLinearAlgebra/LinearMaps.jl/blob/99bbc0e46a196dee503bffee4612bc5494656cd3/src/getindex.jl#L54

Reply:
I'm not sure either how to squeeze in a try-catch block, but I tried and you get the usual "transpose/adjoint not defined for [map type]" error message.